packages/html-reporter/src/reportView.tsx

```diff
@@ -85,19 +85,30 @@ export const ReportView: React.FC<{
         />
       </Route>
       <Route predicate={testCaseRoutePredicate}>
-        {!!report && <TestCaseViewLoader report={report} tests={filteredTests.tests} testIdToFileIdMap={testIdToFileIdMap} />}
+        {!!report ?
+          <TestCaseViewLoader report={report} tests={filteredTests.tests} testIdToFileIdMap={testIdToFileIdMap} /> :
+          <div className='error'>Report data could not be found</div>}
       </Route>
     </main>
   </div>;
 };
 
+type TestState = {
+  type: 'loading'
+} | {
+  type: 'test',
+  test: TestCase,
+} | {
+  type: 'error',
```

Comment:
That would not be a very useful state. How about a placeholder that suggests possible recovery when the report is empty?